Sources/Migration/DataPipelineMigrationAssistant.swift

```diff
@@ -1,54 +1,49 @@
-import CioDataPipelines
 import CioInternalCommon
 import Foundation
 
-public protocol DataPipelineMigration: AutoMockable {
-    func handleAlreadyIdentifiedMigratedUser()
-    func handleQueueBacklog()
+public protocol DataPipelineMigrationAction {
+    func processAlreadyIdentifiedUser(identifier: String)
+    func processIdentifyFromBGQ(identifier: String, timestamp: String, body: [String: Any]?)
+    func processScreenEventFromBGQ(identifier: String, name: String, timestamp: String?, properties: [String: Any])
+    func processEventFromBGQ(identifier: String, name: String, timestamp: String?, properties: [String: Any])
+    func processDeleteTokenFromBGQ(identifier: String, token: String, timestamp: String)
+    func processRegisterDeviceFromBGQ(identifier: String, token: String, timestamp: String, attributes: [String: Any]?)
+    func processPushMetricsFromBGQ(token: String, event: Metric, deliveryId: String, timestamp: String, metaData: [String: Any])
 }
 
-// sourcery: InjectRegister = "DataPipelineMigrationAssistant"
-// sourcery: InjectSingleton
-/// Responsible for handling migration of pending tasks from `Tracking` module to `DataPipeline` module.
-class DataPipelineMigrationAssistant: DataPipelineMigration {
+public class DataPipelineMigrationAssistant {
+    public var migrationHandler: DataPipelineMigrationAction
     private let logger: Logger
     private let backgroundQueue: Queue
     private let jsonAdapter: JsonAdapter
     private let threadUtil: ThreadUtil
     private var profileStore: ProfileStore
 
-    /**
-     Constructor for singleton, only.
-
-     Try loading the credentials previously saved for the singleton instance.
-     */
-    init(
-        logger: Logger,
-        queue: Queue,
-        jsonAdapter: JsonAdapter,
-        threadUtil: ThreadUtil,
-        profileStore: ProfileStore
-    ) {
-        self.logger = logger
-        self.backgroundQueue = queue
-        self.jsonAdapter = jsonAdapter
-        self.threadUtil = threadUtil
-        self.profileStore = profileStore
+    public init(handler: DataPipelineMigrationAction, diGraph: DIGraph) {
```

Comment:
I am not sure if making it dependent on `diGraph: DIGraph` is a good idea, wrapper modules would need to create `DIGraph` then. 

Maybe just rely on `DIGraphShared`?

Reply:
I realised this while writing test cases. Thanks for the heads up. Updating the PR with the change.

Reply:
I am realising it while writing the test cases and the wrappers case is legit too. We get all the values from `DiGraphShared` except `queue` and `profileStore` and that's where I feel blocked.

Reply:
Aman, you might be able to get unblocked after these 2 changes: 
* https://github.com/customerio/customerio-ios/pull/547
* https://github.com/customerio/customerio-ios/commit/dab6fb8c6964848bfaaf0bfe7a6781263cbb4e7a (now in `main-replica-cdp` branch)

The ProfileStore and BQ are now decoupled from `SdkConfig` which I think is why we had to use DIGraph. After these 2 changes above, I think we can now move the BQ and ProfileStore into DIGraphShared.

Reply:
Happy to help. 

I marked the conversation as not resolved because I see the constructor is still using `DIGraph`. Is there a plan to change it?

Reply:
Use of Digraph has been totally removed. The change can be seen in [other PR](https://github.com/customerio/customerio-ios/pull/542)